au4/src/trackedit/internal/au3/au3interaction.cpp

```diff
@@ -628,3 +629,35 @@ void Au3Interaction::pushProjectHistoryJoinState(secs_t start, secs_t duration)
 
     trackeditProject->pushHistoryState(ss.str(), "Join");
 }
+
+void Au3Interaction::undo()
+{
+    auto trackeditProject = globalContext()->currentProject()->trackeditProject();
+    if (!trackeditProject->undoAvailable()) {
+        interactive()->error(muse::trc("undo", "Undo"), std::string("Undo not available"));
```

Comment:
I like how it's done [in MuseScore](https://github.com/musescore/MuseScore/blob/master/src/notation/internal/notationactioncontroller.cpp#L224) - the user can't call undo if there's nothing to undo and therefore this dialog is not needed in MuseScore

Reply:
we can fix this in the next PR